### PR TITLE
Adds the RobotFPS example

### PR DESCRIPTION
**Pull request by Ivan-267**
## RobotFPS environment:
An arcade style FPS environment, trained using SB3 PPO, observations contain 3 stacked frames of data from a raycast sensor (extended to add the facing direction of any visible robot relative to the agent). Making this env took quite a bit of experimentation with different types of obs and rewards while observing the trained behavior.

Trained agent onnx inference:

https://github.com/user-attachments/assets/4e7ecd92-f2e9-49c3-8b75-1a1f67af7922

Todo:
- [x] Add readme
- [x] Add video to repository readme

### CODE REVIEW

**Comment by edbeeching** [comment]
Amazing example @Ivan-267 , I will have time to look at the code later this week.

**Comment by DQN-Labs** [comment]
@Ivan-267 Nice env! I love the idea. Just a suggestion, can you add a few maps instead of just one. I tried the FPS example, and found it quite monotonous. Other than that, amazing environment.

**Comment by Ivan-267** [comment]
> @Ivan-267 Nice env! I love the idea. Just a suggestion, can you add a few maps instead of just one. I tried the FPS example, and found it quite monotonous. Other than that, amazing environment.

Thanks for your feedback and the suggestion! 

The example is meant for users to test and modify/reuse/build upon, so anyone could for example add their own map, different designs or additional gameplay mechanics. I definitely agree it would be fun to have more maps, but at least at the moment it's out of scope of the example (but could be considered for a future time).

File: examples/RobotFPS/scenes/ball/ball.gd [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Reference to the particle effect used on impact
+@onready var impact_effect := $BallImpactEffect
+
+var time_since_spawned := 0.0
+
+
+func _physics_process(delta: float) -> void:
+	# 5 seconds after spawning, the ball gets deleted
+	# to prevent potential accumulation in case a ball doesn't hit anything
+	time_since_spawned += delta
```
Comment (edbeeching) [inline examples/RobotFPS/scenes/ball/ball.gd:16]: It can be neater to use a Timer for this, but not blocking merging

**Review by edbeeching — APPROVED** [review]
Thanks, very clean implementation. One comment but nothing blocking.

File: examples/RobotFPS/scenes/ball/ball.gd (hunk 1 above)
Comment (Ivan-267) [reply examples/RobotFPS/scenes/ball/ball.gd:16]: I used "await timer" initially, but removed it during some memory optimization changes, as the node can be removed before the timer elapses (if the ball hits something). Using the "internal timer", it simply stops counting if `_physics_process` is no longer called.

I'll likely merge the env in a few days or so, to sync with a linkedin post. Thanks for the review!

**Pull request merged**